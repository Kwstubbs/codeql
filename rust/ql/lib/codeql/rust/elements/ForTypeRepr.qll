// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
