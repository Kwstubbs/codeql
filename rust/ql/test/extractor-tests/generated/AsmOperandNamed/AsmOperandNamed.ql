// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

query predicate instances(
  AsmOperandNamed x, string hasAsmOperand__label, string hasAsmOperand, string hasName__label,
  string hasName
) {
  toBeTested(x) and
  not x.isUnknown() and
  hasAsmOperand__label = "hasAsmOperand:" and
  (if x.hasAsmOperand() then hasAsmOperand = "yes" else hasAsmOperand = "no") and
  hasName__label = "hasName:" and
  if x.hasName() then hasName = "yes" else hasName = "no"
}

query predicate getAsmOperand(AsmOperandNamed x, AsmOperand getAsmOperand) {
  toBeTested(x) and not x.isUnknown() and getAsmOperand = x.getAsmOperand()
}

query predicate getName(AsmOperandNamed x, Name getName) {
  toBeTested(x) and not x.isUnknown() and getName = x.getName()
}
