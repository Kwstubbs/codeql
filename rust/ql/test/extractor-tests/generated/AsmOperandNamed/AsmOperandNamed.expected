instances
| gen_asm_operand_named.rs:8:34:8:43 | AsmOperandNamed | hasAsmOperand: | yes | hasName: | no |
| gen_asm_operand_named.rs:8:46:8:62 | AsmOperandNamed | hasAsmOperand: | yes | hasName: | yes |
getAsmOperand
| gen_asm_operand_named.rs:8:34:8:43 | AsmOperandNamed | gen_asm_operand_named.rs:8:34:8:43 | AsmRegOperand |
| gen_asm_operand_named.rs:8:46:8:62 | AsmOperandNamed | gen_asm_operand_named.rs:8:54:8:62 | AsmRegOperand |
getName
| gen_asm_operand_named.rs:8:46:8:62 | AsmOperandNamed | gen_asm_operand_named.rs:8:46:8:50 | input |
